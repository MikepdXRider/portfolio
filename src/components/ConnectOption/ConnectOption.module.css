.img {
  max-height: 50px;
}

.link {
  color: #ffffff;
  font-size: 20px;
  display: flex;
  align-items: center;
  gap: 5px;
}
