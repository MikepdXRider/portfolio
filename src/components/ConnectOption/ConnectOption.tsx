import React from 'react';
// styles
import styles from './ConnectOption.module.css';

interface ConnectOptionProps {
  imgSrc: string;
  imgAlt: string;
  linkHref: string;
  linkText: string;
  linkAriaLabel: string;
  isDownload?: boolean;
}

export default function ConnectOption({
  imgSrc,
  imgAlt,
  linkHref,
  linkText,
  linkAriaLabel,
  isDownload = false,
}: ConnectOptionProps) {
  return (
    <article>
      <a
        href={linkHref}
        aria-label={linkAriaLabel}
        className={styles.link}
        target="_blank"
        rel="noreferrer"
        download={isDownload}
      >
        <img src={imgSrc} alt={imgAlt} className={styles.img} />
        {linkText}
      </a>
    </article>
  );
}
