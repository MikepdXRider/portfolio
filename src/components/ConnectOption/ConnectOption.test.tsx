// testing
import { render, screen } from '@testing-library/react';
// components
import ConnectOption from './ConnectOption';

describe('ConnectOption component', () => {
  const mockContactProps = {
    imgSrc: 'https://placekitten.com',
    imgAlt: 'mock alt',
    linkHref: 'https://placecage.com',
    linkText: 'mock link text',
    linkAriaLabel: 'mock link aria-label',
  };

  it('matches the existing snapshot', () => {
    const { container } = render(<ConnectOption {...mockContactProps} />);

    expect(container).toMatchSnapshot();
  });

  it('matches the existing snapshot WITH download attribute', () => {
    const { container } = render(
      <ConnectOption {...mockContactProps} isDownload />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders component with all passed information', () => {
    render(<ConnectOption {...mockContactProps} />);

    // grabs elements from rendered dom.
    const imgEl = screen.getByRole('img', { name: /mock alt/i });
    const anchorEl = screen.getByLabelText(/mock link aria-label/i);

    //checks that elements are in dom and rendered with correct info.
    expect(imgEl).toBeInTheDocument();
    expect(imgEl).toHaveAttribute('src', 'https://placekitten.com');
    expect(anchorEl).toBeInTheDocument();
    expect(anchorEl).toHaveAttribute('href', 'https://placecage.com');
  });

  it('renders component WITH a download attribute', () => {
    render(<ConnectOption {...mockContactProps} isDownload />);

    // grabs elements from rendered dom.
    const anchorEl = screen.getByLabelText(/mock link aria-label/i);

    //checks that the element is rendered to the dom correctly.
    expect(anchorEl).toHaveAttribute('download');
  });

  it('renders component WITHOUT a download attribute', () => {
    render(<ConnectOption {...mockContactProps} />);

    // grabs elements from rendered dom.
    const anchorEl = screen.getByLabelText(/mock link aria-label/i);

    //checks that the element is rendered to the dom correctly.
    expect(anchorEl).not.toHaveAttribute('download');
  });
});
