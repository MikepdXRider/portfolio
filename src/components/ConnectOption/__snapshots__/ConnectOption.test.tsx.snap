// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectOption component matches the existing snapshot 1`] = `
<div>
  <article>
    <a
      aria-label="mock link aria-label"
      class="link"
      href="https://placecage.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="mock alt"
        class="img"
        src="https://placekitten.com"
      />
      mock link text
    </a>
  </article>
</div>
`;

exports[`ConnectOption component matches the existing snapshot WITH download attribute 1`] = `
<div>
  <article>
    <a
      aria-label="mock link aria-label"
      class="link"
      download=""
      href="https://placecage.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="mock alt"
        class="img"
        src="https://placekitten.com"
      />
      mock link text
    </a>
  </article>
</div>
`;
